### PR TITLE
[Carbondata-379] Scan package's unit test cases added

### PR DESCRIPTION
**Pull request by deepakmehra10**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[CARBONDATA-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).
 - [ ] Testing done
 
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - What manual testing you have done?
        - Any additional information to help reviewers in testing this change.
         
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
                 

---

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -6,9 +6,7 @@
     The ASF licenses this file to You under the Apache License, Version 2.0
     (the "License"); you may not use this file except in compliance with
     the License.  You may obtain a copy of the License at
-
        http://www.apache.org/licenses/LICENSE-2.0
```
Comment (ravipesala) [inline core/pom.xml]: why pom is updated?

File: core/src/main/java/org/apache/carbondata/scan/complextypes/PrimitiveQueryType.java [hunk 2]
```diff
@@ -166,6 +166,7 @@ public PrimitiveQueryType(String name, String parentname, int blockIndex,
       DirectDictionaryGenerator directDictionaryGenerator = DirectDictionaryKeyGeneratorFactory
           .getDirectDictionaryGenerator(dataType);
       actualData = directDictionaryGenerator.getValueFromSurrogate(surrgateValue);
+
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/scan/complextypes/PrimitiveQueryType.java]: Please don't add space

**Comment by ravipesala** [comment]
Please format the classes using formatter available in dev package

**Comment by deepakmehra10** [comment]
Hi Ravi 

Thank you for writing back I will scrap out all the spaces and also format the classes.About POM there were some differences in my Core's POM when I rebased my POM with the master apache/incubator-carbondata ,So I made it exact similar to master

File: core/src/test/java/org/apache/carbondata/scan/filter/FilterUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        assert(result);
+    }
+
+    @Test
+    public void testCheckIfLeftExpressionRequireEvaluationWithExpressionNotInstanceOfColumnExpression(){
+        ColumnExpression expression = new ColumnExpression("test", DataType.STRING);
+        boolean result = FilterUtil.checkIfLeftExpressionRequireEvaluation(expression);
+        assert(!result);
+    }
+
```
Comment (ravipesala) [inline core/src/test/java/org/apache/carbondata/scan/filter/FilterUtilTest.java]: Please cover other methods also inside FilterUtil

File: core/src/test/java/org/apache/carbondata/scan/filter/resolver/metadata/FilterResolverMetadataTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.carbondata.scan.filter.resolver.metadata;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FilterResolverMetadataTest {
```
Comment (ravipesala) [inline core/src/test/java/org/apache/carbondata/scan/filter/resolver/metadata/FilterResolverMetadataTest.java]: Not required testcases for non business logic classes

File: core/src/test/java/org/apache/carbondata/scan/result/impl/NonFilterQueryScannedResultTest.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    public  void setUp(){
+        BlockExecutionInfo blockExecutionInfo = new BlockExecutionInfo();
+        QueryDimension queryDimension[] = {new QueryDimension("dummyColumnName1"),new QueryDimension("dummyColumnName2")};
+        blockExecutionInfo.setQueryDimensions(queryDimension);
+        nonFilterQueryScannedResult = new NonFilterQueryScannedResult(blockExecutionInfo);
+
+    }
+
+    @Test
+    public void testIsNullMeasureValue(){
```
Comment (ravipesala) [inline core/src/test/java/org/apache/carbondata/scan/result/impl/NonFilterQueryScannedResultTest.java]: this test is doing nothing, please mock it properly. We supposed to set `measureDataChunks` and call this method.

File: core/src/test/java/org/apache/carbondata/scan/result/impl/NonFilterQueryScannedResultTest.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        int ordinal = 1;
+        new MockUp<AbstractScannedResult>(){
+            @Mock
+            public boolean isNullMeasureValue(int ordinal, int rowIndex) {
+                return true;
+            }
+        };
+        boolean result = nonFilterQueryScannedResult.isNullMeasureValue(ordinal);
+        assert(result);
+    }
```
Comment (ravipesala) [inline core/src/test/java/org/apache/carbondata/scan/result/impl/NonFilterQueryScannedResultTest.java]: we can add some more testcases to it.

File: core/src/test/java/org/apache/carbondata/scan/result/iterator/RawResultIteratorTest.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        SegmentProperties segmentProperties1 = new SegmentProperties(list,dummyArray);
+        rawResultIterator = new RawResultIterator(detailRawQueryResultIterator,segmentProperties,segmentProperties1);
+    }
+
+    @Test
+    public void testFetchConvertedForNull() throws Exception
+    {
+        Object objects = rawResultIterator.fetchConverted();
+        assertNull(objects);
+    }
```
Comment (ravipesala) [inline core/src/test/java/org/apache/carbondata/scan/result/iterator/RawResultIteratorTest.java]: We can add some more tests to test actual logic

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**